ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -110,6 +118,43 @@ public VcrReplicationManager(CloudConfig cloudConfig, ReplicationConfig replicat
       throw new IllegalStateException("VcrHelixConfig is not correct");
     }
     vcrMetrics.registerVcrHelixUpdateGauge(this::getVcrHelixUpdaterAsCount, this::getVcrHelixUpdateInProgressAsCount);
+
+    // Get the Helix src ZK String for the Helix Sync Operation.
+    // Although only the VCR host which gets assigned partition 1 will do the Helix update,
+    // we get the Helix src ZK string for all the hosts beforehand since there is no harm to do that.
+    // try to use the local DC Zk String first.
+    if (clusterMap instanceof HelixClusterManager) { // Don't run the code if it's Mock cluster.
+      syncSrcClusterName = clusterMapConfig.clusterMapClusterName;
+
+      try {
+        String localDcZkStr = ((HelixClusterManager) clusterMap).getLocalDcZkConnectString();
+        HelixAdmin srcAdmin = new ZKHelixAdmin(localDcZkStr);
+        List<String> srcResources = srcAdmin.getResourcesInCluster(syncSrcClusterName);
+        syncSrcZkStr = localDcZkStr;
+      } catch (Exception e) {
+        logger.warn("Ambry Helix doesn't have a local sync source {} {} ",
+            ((HelixClusterManager) clusterMap).getLocalDcZkConnectString(), syncSrcClusterName, e);
+      }
+
+      // try the remaining ZK strings if local ZK string doesn't work
+      if (syncSrcZkStr == null) {
+        ArrayList<String> zkStrings = ((HelixClusterManager) clusterMap).getAllDcZkConnectionString();
+        logger.info("Ambry Helix zkstring count {} strings {} ", zkStrings.size(), zkStrings);
+
+        for (String zkStr : zkStrings) {
+          try {
+            HelixAdmin srcAdmin = new ZKHelixAdmin(zkStr);
+            List<String> srcResources = srcAdmin.getResourcesInCluster(syncSrcClusterName);
```

Comment:
I am not sure how resource heavy (in terms of data transferred, latency and work done by helix) is the API `getResourcesInCluster`. Can we maybe use something lightweight like `ZkUtil.isClusterSetup()` here, since we aren't really going to use the resources.